components/bl-autocomplete-component/src/helpers/use-options-placement.js

```diff
@@ -1,12 +1,12 @@
 import { useLayoutEffect } from 'react';
 
 export const useOptionsPlacement = props => {
-  const { optionsRef, autocompleteHeight, setOptionsPlacement } = props;
-  
+  const { optionsContainerRef, autocompleteHeight, setOptionsPlacement } = props;
+
   useLayoutEffect(() => {
     const handleScroll = () => {
       const viewPortHeight = window.innerHeight;
-      const { bottom, height } = optionsRef.current.getBoundingClientRect();
+      const { bottom, height } = optionsContainerRef.current?.getBoundingClientRect();
```

Comment:
`optionsContainerRef.current?.getBoundingClientRect()`

I assume the question mark means - sometimes we haven't ref
In that case `bottom` and `height` becomes undefined and all code below looks broken
`const bottomCoordinate = placement === 'top' ? undefined + autocompleteHeight + undefined  : undefined;`

And at some render iterations we create a subscription with the invalid handler, and remove it. That looks to me like unneeded action. Could we just skip iterations where the ref is not initialized?
```
if(ref.current) {
...
}
```

Do you agree?